rllib/algorithms/tests/test_algorithm.py

```diff
@@ -183,6 +182,43 @@ def _has_policies(w):
                 self.assertTrue(pol0.action_space.contains(a))
                 test.stop()
 
+                # After having added 2 policies, try to restore the Algorithm,
```

Comment:
Added test to restore an algo from a checkpoint, but with only a subset of the original policies (that are present in the checkpoint). Takes care of the `policies`dict within the "multiagent" section of the config, etc..